src/messaging/interface/verifier/request.ts

```diff
@@ -0,0 +1,142 @@
+import { randomAsHex } from '@polkadot/util-crypto'
+import { CTypeHash, DidResolveKey, DidUri } from '@kiltprotocol/types'
+import { Credential, Did } from '@kiltprotocol/sdk-js'
+
+import {
+  ISession,
+  ICredentialRequest,
+  IEncryptedMessage,
+  IMessage,
+  IRequestCredential,
+  ISubmitCredential,
+} from '../../../types'
+import { isIRequestCredential, isSubmitCredential } from '../../../utils'
+import { decrypt, encrypt, fromBody } from '../../index'
```

Comment:
Do we want this to come from index or actually the decrypt and encrypt internally?

Reply:
It would help people go through the pathing...